src/app/pages/storage/volumes/volume-import-wizard/volume-import-wizard.component.ts

```diff
@@ -348,6 +350,32 @@ export class VolumeImportWizardComponent implements WizardConfiguration {
       dialogRef.componentInstance.submit();
       dialogRef.componentInstance.success.pipe(untilDestroyed(this)).subscribe(() => {
         dialogRef.close(false);
+
+        if (this.pool) {
+          this.modalService.closeSlideIn();
+          this.modalService.refreshTable();
+          this.ws.call('pool.dataset.query', [[['pool', '=', this.pool]]]).pipe(untilDestroyed(this)).subscribe((datasets) => {
+            for (const dataset of datasets) {
```

Comment:
So if we only care to see if there are any encrypted datasets, this needs to be rewritten to something like:
```
const hasLockedDataset = datasets.some((dataset) => dataset.encrypted && dataset.locked);
```